lib/github/hooks.nit

```diff
@@ -0,0 +1,172 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Github hook event listening with `nitcorn`.
+#
+# Usage:
+#
+# ~~~
+# import github::hooks
+#
+# # A simple hook listener that print received events in stdout.
+# class LogHookListener
+#	super HookListener
+#
+#	redef fun api: GithubAPI do return new GithubAPI(get_github_oauth)
+#
+#	redef fun apply_event(event) do
+#		printn "[{event.repo.full_name}] "
+#		if event isa CommitCommentEvent then
+#			print "new comment on commit {event.comment.commit.sha}"
+#		else if event isa CreateEvent then
+#			print "new {event.ref_type} {event.ref}"
+#		else if event isa DeleteEvent then
+#			print "deleted ref {event.ref_type} {event.ref}"
+#		else if event isa DeploymentEvent then
+#			print "new deployment {event.name}"
+#		else if event isa DeploymentStatusEvent then
+#			print "new deployment state {event.state}"
+#		else if event isa ForkEvent then
+#			print "new fork {event.forkee.full_name}"
+#		else if event isa IssueCommentEvent then
+#			print "new comment on #{event.issue.number}"
+#		else if event isa IssuesEvent then
+#			print "new action {event.action} on #{event.issue.number}"
+#		else if event isa MemberEvent then
+#			print "new member {event.member.login}"
+#		else if event isa PullRequestEvent then
+#			print "new pr event on #{event.number}"
+#		else if event isa PullRequestReviewCommentEvent then
+#			print "new pr comment on #{event.pull.number}"
+#		else if event isa PushEvent then
+#			print "new push event from {event.ref}"
+#		else if event isa StatusEvent then
+#			print "new status {event.state} for {event.commit.sha}"
+#		else
+#			print "unknwon event..."
+#		end
+#	end
+# end
+#
+# var api = new GithubAPI(get_github_oauth)
+# var listener = new LogHookListener(api, "127.0.0.1", 8080)
+# # listener.listen # uncomment to start listening
```

Comment:
Je voulais justifier la ligne en commentaire par autre chose que `# avoid infinite loop in nitunit`